tests/unit/test_mdns.cpp

```diff
@@ -1035,7 +1082,14 @@ static void ParseMessage(const Message &aMessage, const Core::AddressInfo *aUnic
         }
         else
         {
-            msg->mType        = kUnicastResponse;
+            if (aUnicastDest->mPort == kEphemeralPort)
+            {
+                msg->mType = kLegacyUnicastResponse;
+            }
+            else
+            {
+                msg->mType = kUnicastResponse;
+            }
```

Comment:
```suggestion
            msg->mType = (aUnicastDest->mPort == kEphemeralPort) ? LegacyUnicastResponse : kUnicastResponse;
```

Reply:
Will address these in a future push. Thanks @jwhui for the feedback!

Reply:
Changes done as suggested.